tests/components/tasmota/test_light.py

```diff
@@ -0,0 +1,768 @@
+        "Wake up",
+        "Cycle up",
+        "Cycle down",
+        "Random",
+    ]
+    assert state.attributes.get("min_mireds") == 153
+    assert state.attributes.get("max_mireds") == 500
+    assert (
+        state.attributes.get("supported_features")
+        == SUPPORT_BRIGHTNESS
+        | SUPPORT_COLOR
+        | SUPPORT_COLOR_TEMP
+        | SUPPORT_EFFECT
+        | SUPPORT_TRANSITION
+        | SUPPORT_WHITE_VALUE
+    )
+
+
+async def test_attributes_rgbww_reduced(hass, mqtt_mock, setup_tasmota):
+    """Test state update via MQTT."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    config["rl"][0] = 2
+    config["lt_st"] = 5  # 5 channel light (RGBCW)
+    config["so"]["82"] = 1  # Reduced CT range
+    mac = config["mac"]
+
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{mac}/config",
+        json.dumps(config),
+    )
+    await hass.async_block_till_done()
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/LWT", "Online")
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/STATE", '{"POWER":"ON"}')
+
+    state = hass.states.get("light.test")
+    assert state.attributes.get("effect_list") == [
+        "None",
+        "Wake up",
+        "Cycle up",
+        "Cycle down",
+        "Random",
+    ]
+    assert state.attributes.get("min_mireds") == 200
+    assert state.attributes.get("max_mireds") == 380
+    assert (
+        state.attributes.get("supported_features")
+        == SUPPORT_BRIGHTNESS
+        | SUPPORT_COLOR
+        | SUPPORT_COLOR_TEMP
+        | SUPPORT_EFFECT
+        | SUPPORT_TRANSITION
+        | SUPPORT_WHITE_VALUE
+    )
+
+
+async def test_controlling_state_via_mqtt(hass, mqtt_mock, setup_tasmota):
+    """Test state update via MQTT."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    config["rl"][0] = 2
+    config["lt_st"] = 5  # 5 channel light (RGBCW)
+    mac = config["mac"]
+
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{mac}/config",
+        json.dumps(config),
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("light.test")
+    assert state.state == "unavailable"
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/LWT", "Online")
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/STATE", '{"POWER":"ON"}')
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/STATE", '{"POWER":"OFF"}')
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+
+    async_fire_mqtt_message(
+        hass, "tasmota_49A3BC/tele/STATE", '{"POWER":"ON","Dimmer":50}'
+    )
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("brightness") == 127.5
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)